shell/platform/common/accessibility_bridge.cc

```diff
@@ -177,6 +199,59 @@ void AccessibilityBridge::OnAtomicUpdateFinished(
   }
 }
 
+std::optional<ui::AXTreeUpdate>
+AccessibilityBridge::CreateRemoveReparentedNodesUpdate() {
+  std::unordered_map<int32_t, ui::AXNodeData> updates;
+
+  for (auto node_update : pending_semantics_node_updates_) {
+    for (int32_t child_id : node_update.second.children_in_traversal_order) {
+      // Skip nodes that don't exist or have a parent in the current tree.
+      ui::AXNode* child = tree_.GetFromId(child_id);
+      if (!child || !child->parent()) {
+        continue;
+      }
+
+      // Skip nodes whose parents are unchanged.
+      if (child->parent()->id() == node_update.second.id) {
+        continue;
+      }
+
+      // This pending update moves the current child node.
+      // That new child must have a corresponding pending update.
+      assert(pending_semantics_node_updates_.find(child_id) !=
+             pending_semantics_node_updates_.end());
+
+      // Create an update to remove the child from its previous parent.
+      int32_t parent_id = child->parent()->id();
```

Comment:
Do we need special handling for the root node? My recollection is that we ourselves synthesize the root node so it's never removed; is that correct?

Reply:
The `AXTree` accepts updates that moves the root node. It's only when a node's parents change that `AXTree` rejects a "move".

Reply:
> My recollection is that we ourselves synthesize the root node so it's never removed; is that correct?

That's correct, there's always a root node with id 0: https://github.com/flutter/flutter/blob/f852c46c877bc321c1705403cfb2b94e5976de3f/packages/flutter/lib/src/rendering/object.dart#L4112-L4115

Also, while the underlying `AXTree` supports changing the root node (using `AxTreeUpdateBase.root_id`), the `AccessibilityBridge` never uses that.

I'll add an assertion that the root node is never moved.

Reply:
Added an assertion!